test/e2e/windows/gmsa_full.go

```diff
@@ -165,53 +181,10 @@ var _ = SIGDescribe("[Feature:Windows] GMSA Full [Serial] [Slow]", func() {
 		})
 
 		ginkgo.It("can read and write file to remote SMB folder", func() {
-			defer ginkgo.GinkgoRecover()
-
-			ginkgo.By("finding the worker node that fulfills this test's assumptions")
-			nodes := findPreconfiguredGmsaNodes(f.ClientSet)
-			if len(nodes) != 1 {
-				e2eskipper.Skipf("Expected to find exactly one node with the %q label, found %d", gmsaFullNodeLabel, len(nodes))
-			}
-			node := nodes[0]
-
-			ginkgo.By("retrieving the contents of the GMSACredentialSpec custom resource manifest from the node")
-			crdManifestContents := retrieveCRDManifestFileContents(f, node)
-
-			ginkgo.By("deploying the GMSA webhook")
-			webhookCleanUp, err := deployGmsaWebhook(f)
-			defer webhookCleanUp()
-			if err != nil {
-				framework.Failf(err.Error())
-			}
-
-			ginkgo.By("creating the GMSA custom resource")
-			customResourceCleanup, err := createGmsaCustomResource(f.Namespace.Name, crdManifestContents)
-			defer customResourceCleanup()
-			if err != nil {
-				framework.Failf(err.Error())
-			}
-
-			ginkgo.By("creating an RBAC role to grant use access to that GMSA resource")
-			rbacRoleName, rbacRoleCleanup, err := createRBACRoleForGmsa(f)
-			defer rbacRoleCleanup()
-			if err != nil {
-				framework.Failf(err.Error())
-			}
-
-			ginkgo.By("creating a service account")
-			serviceAccountName := createServiceAccount(f)
-
-			ginkgo.By("binding the RBAC role to the service account")
-			bindRBACRoleToServiceAccount(f, serviceAccountName, rbacRoleName)
-
-			ginkgo.By("creating a pod using the GMSA cred spec")
-			podName := createPodWithGmsa(f, serviceAccountName)
-
 			ginkgo.By("getting the ip of GMSA domain")
-			gmsaDomainIP := getGmsaDomainIP(f, podName)
 
 			ginkgo.By("checking that file can be read and write from the remote folder successfully")
-			filePath := fmt.Sprintf("\\\\%s\\%s\\write-test-%s.txt", gmsaDomainIP, gmsaSharedFolder, string(uuid.NewUUID())[0:4])
+			filePath := fmt.Sprintf("\\\\%s\\%s\\write-test-%s.txt", gmsaDomain, gmsaSharedFolder, string(uuid.NewUUID())[0:4])
```

Comment:
Should we check that we can read/write from both the IP and domain name?

Reply:
I think it makes sense to have both checked since we had the bug with dns names but ip worked. With this refactoring it should be easy enough to add another test case such as `ginkgo.It("can read and write file to remote SMB folder by ip"`

Reply:
Hi @marosset and @jsturtevant, it makes sense to me to add both test cases. But I find that I can't access the file using domain name. 
```
$ /home/kubo/path/kubectl --kubeconfig=/home/kubo/.kube/config --namespace=gmsa-full-test-windows-2603 exec --namespace=gmsa-full-test-windows-2603 pod-with-gmsa powershell.exe -- cat \\\\k8sgmsa.lan\\write_test\\write-test-0c0d.txt
cat: can't open '\\k8sgmsa.lan\write_test\write-test-0c0d.txt': No such file or directory
command terminated with exit code 1
$ 
$ 
$ /home/kubo/path/kubectl --kubeconfig=/home/kubo/.kube/config --namespace=gmsa-full-test-windows-2603 exec --namespace=gmsa-full-test-windows-2603 pod-with-gmsa powershell.exe -- cat \\\\10.180.127.203\\write_test\\write-test-0c0d.txt
This is a test file.
$ 
$ /home/kubo/path/kubectl --kubeconfig=/home/kubo/.kube/config --namespace=gmsa-full-test-windows-2603 exec --namespace=gmsa-full-test-windows-2603 pod-with-gmsa powershell.exe -- ping k8sgmsa.lan

Pinging k8sgmsa.lan [10.180.127.203] with 32 bytes of data:
Reply from 10.180.127.203: bytes=32 time=2ms TTL=127
Reply from 10.180.127.203: bytes=32 time=1ms TTL=127
Reply from 10.180.127.203: bytes=32 time<1ms TTL=127
Reply from 10.180.127.203: bytes=32 time<1ms TTL=127

Ping statistics for 10.180.127.203:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 0ms, Maximum = 2ms, Average = 0ms
``` 
I'm using AD 2019 and I remember that this issue is fixed in this version. Really appreciate any hints on how to triage it. Thanks!